scalar/nil_test.go

```diff
@@ -0,0 +1,18 @@
+package scalar
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsNil(t *testing.T) {
+	s := make([]*string, 1)
+	s[0] = nil
+	require.True(t, IsNil(s[0]))
+	require.False(t, isNilTrivial(s[0]))
```

Comment:
Seems like we could make this a table-driven test with some more types and assertions?

Reply:
Also line 13 seems to be testing the test itself 🙃

Reply:
Added a test in 346bd33

Reply:
do we want to?